src/Blog/Infrastructure/OpenGraph/ImagickImageGenerator.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Blog\Infrastructure\OpenGraph;
+
+use App\Blog\Domain\Model\Article;
+use App\Blog\Domain\OpenGraph\ImageGenerator;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+final readonly class ImagickImageGenerator implements ImageGenerator
+{
+    public function __construct(
+        #[Autowire(param: 'kernel.project_dir')]
+        private string $projectDir,
+    ) {
+        if (!class_exists(\Imagick::class)) {
+            throw new \RuntimeException('Imagick is not installed');
+        }
+    }
+
+    public function generate(Article $article): string
+    {
+        $image = new \Imagick();
```

Comment:
Finally, I went with plain `Imagick` instead of `intervention/image`. 

The library is awesome, but I needed an imagick specific feature for text kerning (it was possible to use it through `intervention/image`, but it was kind of ugly and anti-pattern). Also, we are not doing really hard complexe things here, I believe it's fine.